File: src/lib/pjecalc/core/comum/validators/limited-to-100-years.ts
/**
 * PJe-Calc v2.15.1 — LimitedTo100Years (annotation TS-adaptada)
 * Porte de: br.jus.trt8.pjecalc.negocio.comum.validators.LimitedTo100Years
 */
export interface LimitedTo100YearsSpec {
  groups?: number[];
  condition?: string;
  message?: string; // default "{MSG0011}"
}

export interface LimitedTo100Years_ {
  groups: number[];
  condition: string;
  message: string;
}

export function limitedTo100Years(spec: LimitedTo100YearsSpec = {}): LimitedTo100Years_ {
  return {
    groups: spec.groups ?? [],
    condition: spec.condition ?? '',
    message: spec.message ?? '{MSG0011}',
  };
}

File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/comum/validators/LimitedTo100Years.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  org.hibernate.validator.ValidatorClass
 */
package br.jus.trt8.pjecalc.negocio.comum.validators;

import br.jus.trt8.pjecalc.negocio.comum.validators.LimitedTo100YearsValidator;
import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.hibernate.validator.ValidatorClass;

@Target(value={ElementType.FIELD})
@Retention(value=RetentionPolicy.RUNTIME)
@ValidatorClass(value=LimitedTo100YearsValidator.class)
@Documented
public @interface LimitedTo100Years {
    public byte[] groups() default {};

    public String condition() default "";

    public String message() default "{MSG0011}";
}

